query-engine/query-engine/src/state.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{context::PrismaContext, logger::Logger, opt::PrismaOpt, PrismaResult};
+use psl::PreviewFeature;
+use query_core::schema::QuerySchemaRef;
+use query_engine_metrics::{setup as metric_setup, MetricRegistry};
+use std::sync::Arc;
+use tracing::Instrument;
+
+//// Shared application state.
+pub struct State {
+    pub cx: Arc<PrismaContext>,
+    pub enable_playground: bool,
+    pub enable_debug_mode: bool,
+    pub enable_metrics: bool,
+}
+
+impl State {
+    /// Create a new instance of `State`.
+    fn new(cx: PrismaContext, enable_playground: bool, enable_debug_mode: bool, enable_metrics: bool) -> Self {
```

Comment:
No need to change this now, but boolean params are very poor in information when you read callers and really easy to get wrong (which boolean goes at which position). In my experience bitflags with enumflags are a lot easier to read when you construct them (all values are named), on top of being more compact in memory (here it would be 1 byte instead of 3). They're also more code and from a library, so, tradeoffs.

Reply:
I totally agree, grabed the whole State object from a previous patch in progress, will try to address in here.

Reply:
Not blocking, like the rest of my feedback here, so do what you feel most comfortable with. No need to learn a whole (small) library for this if you haven't used `enumflags2` before.

Reply:
I like the approach of using enum2flags for modeling a compacted set of flags. FWIW, these do already exist as boolean flags coming from the CLI. so we would have to make a transformation. Nevertheless, I don't want to add yet another boolean to this struct in case we have more flags, so I'm ok with exploring this option. Thanks for the suggestion.

I will do it, though, in a different PR after I merge both this and https://github.com/prisma/prisma-engines/pull/3618 to prevent more conflicts, as the https://github.com/prisma/prisma-engines/pull/3618 has changes using the state.